sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/tests/prerequisites/setup.ps1

```diff
@@ -0,0 +1,183 @@
+# IMPORTANT: Use the contents of this file as a template on what resources should be created in order run the tests
+
+#-tsIDs series.id, id -environmentName Zeta -pathToParams .\params.json -pathToTemplate .\deploy.json
+
+# This script is used to create the necessary resources required to run the unit/E2E tests
+
+param(
+    [Parameter(Mandatory)]
+    [string] $Region,
+
+    [Parameter(Mandatory)]
+    [string] $ResourceGroup,
+    
+    [Parameter(Mandatory)]
+    [string] $SubscriptionId,
+
+    [Parameter(Mandatory)]
+    [string[]] $TimeSeriesIds,
+
+    [Parameter(Mandatory)]
+    [ValidateLength(6, 50)]
+    [string] $EnvironmentName,
+
+    [Parameter(Mandatory)]
+    [string] $ConsumerGroupName,
+
+    [Parameter(Mandatory)]
+    [string] $Timestamp
+)
+
+$user = $env:UserName
+$pathToUserParamsFile = "$PSScriptRoot\params.$user.json"
+
+Function Connect-AzureSubscription
+{
+    # Ensure the user is logged in
+    try
+    {
+        $azureContext = az account show
+    }
+    catch { }
+
+    if (-not $azureContext)
+    {
+        Write-Host "`nPlease login to Azure..."
+        az login
+        $azureContext = az account show
+    }
+
+    # Ensure the desired subscription is selected
+    $sub = az account show --output tsv --query id
+    if ($sub -ne $SubscriptionId)
+    {
+        Write-Host "`nSelecting subscription $SubscriptionId"
+        az account set --subscription $SubscriptionId
+    }
+
+    return $azureContext
+}
+
+Function Write-Params($applicationOId)
+{
+    $tsIDArray = @()
+
+    foreach ($id in $TimeSeriesIds)
+    {
+        $tsId = [pscustomobject]@{name=$id ;type='string'}
+        $tsIDArray += $tsID
+    }
+
+    $pathToTemplateParamsFile = "$PSScriptRoot\params.template.json"
+
+    $a = Get-Content $pathToTemplateParamsFile | ConvertFrom-Json
+    $a.parameters.environmentTimeSeriesIdProperties.value = $tsIDArray
+    $a.parameters.iotHubName.value = $EnvironmentName + "-hub"
+    $a.parameters.environmentName.value = $EnvironmentName
+    $a.parameters.eventSourceName.value = $EnvironmentName + "eventSource"
+    $a.parameters.consumerGroupName.value = $ConsumerGroupName
+    $a.parameters.eventSourceTimestampPropertyName.value = $Timestamp
+    $a.parameters.testApplicationOid.value = $applicationOId
+    $a.parameters.region.value = $Region
+    $a.parameters.resourceGroup.value = $ResourceGroup
+    $a.parameters.subscriptionId.value = $SubscriptionId
+
+    $a | ConvertTo-Json -depth 32 | set-content $pathToUserParamsFile
```

Comment:
So this overwrites the params file? Won't that be seen as something to change in git and be easily, accidentally included in a PR?

Reply:
Oh, I see this is a user-specific params file.

I'm not sure if there is any value in saving this file to disk to use for a next time. Previously we've just done the params in memory. I'm curious why we've gone this direction this time.

Reply:
The params file is used as input for the 'az deployment group create' command. You are correct that we can pass parameters directly inline to the az command. However, because there are many parameters to be passed, I thought it would be cleaner this way. I also chose to keep the user params file around as a reference for the user to go back to if needed. But as you mentioned, doesn't need to be kept on disk. 

@drwill-ms Let me know your thoughts, as I am ok with removing the file from disk.